lib/elastic_apm/opentracing.rb

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+require 'opentracing'
+
+module ElasticAPM
+  module OpenTracing
+    # @api private
+    class Span
+      def initialize(elastic_span)
+        @elastic_span = elastic_span
+      end
+
+      attr_reader :elastic_span
+
+      def context
+        elastic_span.trace_context
+      end
+
+      def finish(end_time: Time.now)
+        elastic_span.done end_time: Util.micros(end_time)
+
+        case elastic_span
+        when ElasticAPM::Transaction
+          ElasticAPM.end_transaction
+        when ElasticAPM::Span
+          ElasticAPM.end_span
+        end
+      end
+    end
+
+    # @api private
+    class SpanContext
+      def initialize(trace_context)
```

Comment:
I am a little bit confused about https://github.com/opentracing/opentracing-ruby/blob/master/lib/opentracing/span_context.rb, but it looks like two different initializers.